### ISSUE TITLE
Navigation Timeout on generating pdf

### ISSUE THREAD

**Issue by atif** [Studiosity/grover#255]
I'm getting a navigation timeout randomly (majorly if I refresh the page). I have a rails end point that returns a pdf generated by Grover (to_pdf). It works fine on first page load but if I refresh the page (it kinda gets unresponsive and results in a navigation timeout). 

Also, if change some contents of the html.erb and refresh the page, i get navigation timeout.

Any idea about how to fix this?

**Comment by abrom** [comment]
You're going to have to provide a little context around how you're actually using Grover. What have you tried in the way of [debugging](https://github.com/Studiosity/grover?tab=readme-ov-file#debugging) ??

**Comment by itsmegrave** [comment]
I think I have a similiar issue here.
Using the debugging variables, I found this error:
```
(irb):1:in `<main>': Failed to launch the browser process! undefined (Grover::JavaScript::Error)
[31:49:0912/143900.690452:ERROR:bus.cc(407)] Failed to connect to the bus: Failed to connect to socket /var/run/dbus/system_bus_socket: No such file or directory
[31:31:0912/143900.710846:ERROR:ozone_platform_x11.cc(244)] Missing X server or $DISPLAY
[31:31:0912/143900.710930:ERROR:env.cc(258)] The platform failed to initialize.  Exiting.
```

Using Ruby 3.1.6, Puppeteer 22.13.1 and Grover 1.1.9, inside a docker container.

**Comment by abrom** [comment]
No, this does not appear to be a similar issue. The error message is telling you what your problem is. 
```
Missing X server or $DISPLAY
```
LMGTFY: https://github.com/puppeteer/puppeteer/issues/8148

**Comment by Arns** [comment]
@atif do you have the display_url set? That's what fixed my issues. Ex: 
`display_url: Rails.configuration.x.grover.base_url,`
